app/src/main/java/com/eis/phoneringer/structure/ReceivedMessageListener.java

```diff
@@ -0,0 +1,56 @@
+package com.eis.phoneringer.structure;
+
+import android.content.Context;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+
+import com.eis.phoneringer.exceptions.IllegalPasswordException;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.listeners.SMSReceivedListener;
+
+/**
+ * Listener used to manage incoming SMS and evaluate if they are valid commands for the application
+ *
+ * @author Alberto Ursino, Luca Crema
+ */
+public class ReceivedMessageListener implements SMSReceivedListener {
+
+    private RingtoneHandler ringtoneHandler = RingtoneHandler.getInstance();
+    private Context context;
+
+    /**
+     * Constructor which define a context
+     *
+     * @param context The current application context
+     */
+    public ReceivedMessageListener(Context context) {
+        this.context = context;
+    }
+
+    /**
+     * Creates the RingCommand object upon receipt of the message and calls the class {@link AppManager} in order to manage it only if it isn't null
```

Comment:
"in order to manage" what? The RingCommand object or the message? I would specify what I'm referring to instead of using "it", repetitions are not a problem in documentation if they help make it more clear

Reply:
I'll specify better.